pulp_container/app/registry_api.py

```diff
@@ -1009,3 +1051,75 @@ def get_response_data(signatures):
             }
             data.append(signature)
         return {"signatures": data}
+
+    def put(self, request, path, pk):
+        """Create a new signature from the received data."""
+        _, repository = self.get_dr_push(request, path)
+
+        try:
+            manifest = models.Manifest.objects.get(
+                digest=pk, pk__in=repository.latest_version().content
+            )
+        except models.Manifest.DoesNotExist:
+            raise ManifestNotFound(reference=pk)
+
+        signature_payload = request.META["wsgi.input"].read(SIGNATURE_PAYLOAD_MAX_SIZE)
+        try:
+            signature_dict = json.loads(signature_payload)
+        except json.decoder.JSONDecodeError:
+            raise ManifestSignatureInvalid(digest=pk)
+
+        serializer = serializers.ManifestSignaturePutSerializer(data=signature_dict)
+        serializer.is_valid(raise_exception=True)
+
+        try:
+            signature_raw = base64.b64decode(signature_dict["content"])
+        except binascii.Error:
+            raise ManifestSignatureInvalid(digest=pk)
+
+        signature_json = extract_data_from_signature(signature_raw, manifest.digest)
+        if signature_json is None:
+            raise ManifestSignatureInvalid(digest=pk)
+
+        sig_digest = hashlib.sha256(signature_raw).hexdigest()
+        signature = models.ManifestSignature(
+            name=f"{manifest.digest}@{sig_digest[:32]}",
```

Comment:
please do like here https://github.com/pulp/pulp_container/pull/533/files#diff-b702cbd3ebdff19bf8e4dfc14688aa3fb36e1a47ef6cf84d24e0f75b61247d36R365

Reply:
i have not seen the serializer for this yet, so feel free to ignore my previous comment